### PR TITLE
Made the getByteLengthFunction use MVVM pattern in a more MVVM way. 

### PR DESCRIPTION
**Pull request by Landrito**
Transformation methods and objects should only be accessed by the transformers. The surfaces should only be able to handle view objects.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java [hunk 1]
```diff
@@ -498,7 +484,7 @@ public String getByteLengthFunction(Interface service, Method method, TypeRef ty
     switch (typeRef.getKind()) {
       case TYPE_MESSAGE:
         return "gax.createByteLengthFunction(grpcClients."
-            + getStubNameFor(service, method)
+            + getApiMethodView(service, method).stubName()
```
Comment (jmuk) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: This was a typo -- it turned out to be `grpcClientVariableName()`. Are there good name for that?

**Comment by Landrito** [comment]
Let me look into this later tonight. I might make a separate view for this
since its pretty inefficient create a whole api method view for a single
name.

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java (hunk 1 above)
Comment (Landrito) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: Fixed.

**Comment by Landrito** [comment]
PTAL. Fixed to use grpcClientVariableName.

File: src/main/java/com/google/api/codegen/transformer/GrpcStubTransformer.java [hunk 2]
```diff
@@ -62,4 +64,23 @@
...
+    stub.fullyQualifiedType(namer.getFullyQualifiedStubType(targetInterface));
+    stub.createStubFunctionName(namer.getCreateStubFunctionName(targetInterface));
+    String grpcClientTypeName = namer.getGrpcClientTypeName(targetInterface);
+    stub.grpcClientTypeName(context.getTypeTable().getAndSaveNicknameFor(grpcClientTypeName));
+    stub.grpcClientVariableName(namer.getGrpcClientVariableName(targetInterface));
+    stub.grpcClientImportName(namer.getGrpcClientImportName(targetInterface));
+    stub.methodNames(Collections.singletonList(namer.getApiMethodName(method)));
+    stub.stubMethodsArrayName(namer.getStubMethodsArrayName(targetInterface));
+    stub.namespace(namer.getNamespace(targetInterface));
+    stub.protoFileName(targetInterface.getFile().getSimpleName());
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/GrpcStubTransformer.java]: This is duplicating the code in the function above.

**Comment by garrettjonesgoogle** [comment]
LGTM

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java [hunk 3]
```diff
@@ -46,6 +46,9 @@
 import com.google.common.collect.ImmutableSet;
 import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
```
Comment (jmuk) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: nit: strip away those imports? seems not used in the new code.

Comment (Landrito) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: Just pushed an amendment. Good catch.

**Pull request merged**